matrix_client/crypto/olm_device.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+import olm
+from canonicaljson import encode_canonical_json
+
+from matrix_client.checks import check_user_id
+
+logger = logging.getLogger(__name__)
+
+
+class OlmDevice(object):
+    """Manages the Olm cryptographic functions.
+
+    Has a unique Olm account which holds identity keys.
+
+    Args:
+        api (MatrixHttpApi): The api object used to make requests.
+        user_id (str): Matrix user ID. Must match the one used when logging in.
+        device_id (str): Must match the one used when logging in.
+    """
+
+    _signing_alg = 'ed25519'
```

Comment:
I didn't want to hardcode this string everywhere. Similarly, curve25519 will be _encryption_alg in later commits. The terminology is not perfect so I'm not entirely sure about this...